### DISCUSSION TITLE
Question about why return 204 NO_Content when doing update

### DISCUSSION THREAD

**Discussion by webswift-engineering** [yahoo/elide#3007] [Q&A] [answered]
Hi,
I am using the Elide api for update. and I noticed that the update request will always return HttpStatus: 204 No Content. Body:  null
Is there a way to make the update api return the data that is being updated?

## Expected Behavior

Given (update data-type/1)
`curl --location --request PATCH 'http://localhost:9091/json-api/data-type/1' \
--header 'Content-Type: application/vnd.api+json' \
--header 'ApiVersion: v1.0' \
--header 'Authorization: c94856c7-50ac-4fd3-b395-89af4c7d7bbc' \
--header 'Accept-Language: en' \
--data-raw '{
    "data": {
        "id": 1,
        "type": "data-type",
        "attributes": {
            "baseUrl": "http://url"
        }
    }
}
'`

Return (Body after updating):
`{
    "data": {
        "id": 1,
        "type": "data-type",
        "attributes": {
            "baseUrl": "http://url"
        }
    }
}`

## Current Behavior

Given (update data-type/1)
`curl --location --request PATCH 'http://localhost:9091/json-api/data-type/1' \
--header 'Content-Type: application/vnd.api+json' \
--header 'ApiVersion: v1.0' \
--header 'Authorization: c94856c7-50ac-4fd3-b395-89af4c7d7bbc' \
--header 'Accept-Language: en' \
--data-raw '{
    "data": {
        "id": 1,
        "type": "data-type",
        "attributes": {
            "baseUrl": "http://url"
        }
    }
}
'`

Return (Body after updating):
null

## Possible Solution

when construct the patch response, is it possible to let end user decide to return response or not?

https://github.com/yahoo/elide/blob/309f35b792124bfbef9cc0bb504b49d2f49f425a/elide-core/src/main/java/com/yahoo/elide/jsonapi/parser/state/BaseState.java#L177

## Context

I am in a context of building a CRUD admin api. And I need to present the response (entity after updating)  to client.

Although I have the option to call /Get api after updating, reading the response after update would make work easier. Because I don't need to call extra apis.

## Your Environment

* Elide version used:
* Environment name and version (Java 1.8.0_152):
* Operating System and version:
* Link to your project:

**Comment by moizarafat** [comment, marked as answer]
you can use this method while building the ElideSettings. https://github.com/yahoo/elide/blob/master/elide-core/src/main/java/com/yahoo/elide/ElideSettingsBuilder.java#L182

**Comment by webswift-engineering** [comment]
Thank you, that is really helpful.

 I think it would be nice to add how to override ElideSetting 

https://github.com/yahoo/elide/blob/309f35b792124bfbef9cc0bb504b49d2f49f425a/elide-spring/elide-spring-boot-autoconfigure/README.md#elide-override

into the Elide documentation

https://elide.io/pages/guide/v6/01-start.html

**Reply by sushimsaini** [reply]
@pagexv  could you please help me out how did you overside ElideSetting . What changes you made.

**Reply by justin-tay** [reply]
Currently in order to change the behavior you need to create a fully configured `RefreshableElide` bean and set `withUpdate200Status` for the settings.

```java
package example;

import java.util.TimeZone;

import org.apache.commons.lang3.StringUtils;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import com.yahoo.elide.Elide;
import com.yahoo.elide.ElideSettingsBuilder;
import com.yahoo.elide.RefreshableElide;
import com.yahoo.elide.core.TransactionRegistry;
import com.yahoo.elide.core.audit.Slf4jLogger;
import com.yahoo.elide.core.datastore.DataStore;
import com.yahoo.elide.core.dictionary.EntityDictionary;
import com.yahoo.elide.core.exceptions.ErrorMapper;
import com.yahoo.elide.core.filter.dialect.RSQLFilterDialect;
import com.yahoo.elide.jsonapi.JsonApiMapper;
import com.yahoo.elide.jsonapi.links.DefaultJSONApiLinks;
import com.yahoo.elide.spring.config.ElideConfigProperties;
import com.yahoo.elide.utils.HeaderUtils;

@Configuration
public class ElideConfiguration {
	@Bean
	public RefreshableElide refreshableElide(EntityDictionary dictionary, DataStore dataStore,
            HeaderUtils.HeaderProcessor headerProcessor, TransactionRegistry transactionRegistry,
            ElideConfigProperties settings, JsonApiMapper mapper, ErrorMapper errorMapper) {

        ElideSettingsBuilder builder = new ElideSettingsBuilder(dataStore)
                .withUpdate200Status()
                .withEntityDictionary(dictionary)
                .withErrorMapper(errorMapper).withJsonApiMapper(mapper)
                .withDefaultMaxPageSize(settings.getMaxPageSize()).withDefaultPageSize(settings.getPageSize())
                .withJoinFilterDialect(RSQLFilterDialect.builder().dictionary(dictionary).build())
                .withSubqueryFilterDialect(RSQLFilterDialect.builder().dictionary(dictionary).build())
                .withAuditLogger(new Slf4jLogger()).withBaseUrl(settings.getBaseUrl())
                .withISO8601Dates("yyyy-MM-dd'T'HH:mm'Z'", TimeZone.getTimeZone("UTC"))
                .withHeaderProcessor(headerProcessor);

        if (settings.isVerboseErrors()) {
            builder.withVerboseErrors();
        }

        if (settings.getAsync() != null && settings.getAsync().getExport() != null
                && settings.getAsync().getExport().isEnabled()) {
            builder.withExportApiPath(settings.getAsync().getExport().getPath());
        }

        if (settings.getGraphql() != null && settings.getGraphql().isEnabled()) {
            builder.withGraphQLApiPath(settings.getGraphql().getPath());

            if (settings.getGraphql().getFederation().isEnabled()) {
                builder.withGraphQLFederation(true);
            }
        }

        if (settings.getJsonApi() != null && settings.getJsonApi().isEnabled()) {
            builder.withJsonApiPath(settings.getJsonApi().getPath());

            if (settings.getJsonApi().getLinks().isEnabled()) {
                String baseUrl = settings.getBaseUrl();

                if (StringUtils.isEmpty(baseUrl)) {
                    builder.withJSONApiLinks(new DefaultJSONApiLinks());
                } else {
                    String jsonApiBaseUrl = baseUrl + settings.getJsonApi().getPath() + "/";
                    builder.withJSONApiLinks(new DefaultJSONApiLinks(jsonApiBaseUrl));
                }
            }
        }

        Elide elide = new Elide(builder.build(), transactionRegistry, dictionary.getScanner(), true);

        return new RefreshableElide(elide);
    }
}
```
